### PR TITLE
Add check for lossy params in source

### PR DESCRIPTION
**Pull request by kkrik-es**
Lossy params may cause issues with reindexing data under the hood. 

The current logic requires an extra setting to enable checking for lossy params that is not currently available.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java [hunk 1]
```diff
@@ -166,6 +170,23 @@ public SourceFieldMapper build() {
...
+                if (enabled.get().value() == false) {
+                    disallowed.add("enabled");
+                }
+                if (includes.get().isEmpty() == false) {
+                    disallowed.add("includes");
+                }
+                if (excludes.get().isEmpty() == false) {
+                    disallowed.add("excludes");
+                }
+                assert disallowed.isEmpty() == false;
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java:184]: I think `mode` option `disabled` should be disallowed as well?

File: server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java [hunk 2]
```diff
@@ -166,6 +170,23 @@ public SourceFieldMapper build() {
             }
             if (isDefault()) {
                 return indexMode == IndexMode.TIME_SERIES ? TSDB_DEFAULT : DEFAULT;
+            } else if (supportsNonDefaultParameterValues == false) {
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java:173]: maybe just an `if` instead of a `else if`?

**Review by martijnvg — APPROVED** [review]
Maybe setting `mode` to `disabled` should be disallowed as well?
Otherwise LGTM

File: server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java (hunk 1 above)
Comment (kkrik-es) [reply server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java:188]: Indeed, added to the check for default mapper too.

**Pull request merged**